vpr/src/route/rr_graph_util.cpp

```diff
@@ -75,3 +80,76 @@ int seg_index_of_sblock(int from_node, int to_node) {
         return OPEN; //Should not reach here once thrown
     }
 }
+
+void reorder_rr_graph_nodes(const t_router_opts& router_opts) {
+    auto& device_ctx = g_vpr_ctx.mutable_device();
+    auto& graph = device_ctx.rr_nodes;
+    size_t v_num = graph.size();
+
+    if (router_opts.reorder_rr_graph_nodes_algorithm == DONT_REORDER) return;
+    if (router_opts.reorder_rr_graph_nodes_threshold < 0 || v_num < (size_t)router_opts.reorder_rr_graph_nodes_threshold) return;
+
+    vtr::ScopedStartFinishTimer timer("Reordering rr_graph nodes");
+
+    vtr::vector<RRNodeId, RRNodeId> src_order(v_num); // new id -> old id
+    size_t cur_idx = 0;
+    for (RRNodeId& n : src_order) { // Initialize to [0, 1, 2 ...]
+        n = RRNodeId(cur_idx++);
+    }
+
+    if (router_opts.reorder_rr_graph_nodes_algorithm == DEGREE_BFS) {
```

Comment:
Should explain what this algorithm is.  Sort first by node out-degree, breaking ties by bfs level from an arbitrarily selected (starting at node 0)?  Or something else?  Also explain the intuition behind the algorithm (why might this be good) and that it currently seems to get the same runtime and memory footprint as the original rr-graph creation order for generated rr-graphs.  Not sure for Artix7 graphs (if you know that worth saying that too).

Reply:
Done.

I mentioned that the `DEGREE_BFS` algorithm could likely be improved. It's just not a priority for me right now, and I'd rather provide a good starting point for whoever has time to do the research, maybe later me.